packages/infolists/src/Components/Concerns/HasIconColor.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Filament\Infolists\Components\Concerns;
+
+use Closure;
+use Filament\Infolists\Components\Component;
+
+trait HasIconColor
+{
+    /**
+     * @var string | array{50: string, 100: string, 200: string, 300: string, 400: string, 500: string, 600: string, 700: string, 800: string, 900: string, 950: string} | bool | Closure | null
+     */
+    protected string | array | bool | Closure | null $iconColor = null;
+
+    /**
+     * @param  string | array{50: string, 100: string, 200: string, 300: string, 400: string, 500: string, 600: string, 700: string, 800: string, 900: string, 950: string} | bool | Closure | null  $color
+     */
+    public function iconColor(string | array | bool | Closure | null $color): static
+    {
+        $this->iconColor = $color;
+
+        return $this;
+    }
+
+    /**
+     * @param  array<mixed> | Closure  $colors
+     */
+    public function iconColors(array | Closure $colors): static
```

Comment:
I don't think we need this API, do we, @danharrin?

Reply:
Agree

Reply:
@zepfietje @danharrin  in my case, I need to make icon color base on record value